src/test/integration/users/AVS.t.sol

```diff
@@ -0,0 +1,160 @@
+    ) public createSnapshot returns (OperatorSet[] memory operatorSets) {
+        print.method("createOperatorSets");
+
+        uint256 len = strategies.length;
+        CreateSetParams[] memory p = new CreateSetParams[](len);
+        operatorSets = new OperatorSet[](len);
+
+        for (uint256 i; i < len; ++i) {
+            p[i] = CreateSetParams({operatorSetId: totalOperatorSets++, strategies: strategies[i]});
+
+            operatorSets[i] = OperatorSet(address(this), p[i].operatorSetId);
+        }
+
+        print.createOperatorSets(p);
+        allocationManager.createOperatorSets(p);
+    }
+
+    function createOperatorSet(
+        IStrategy[] memory strategies
+    ) public createSnapshot returns (OperatorSet memory operatorSet) {
+        print.method("createOperatorSets");
+
+        operatorSet = OperatorSet(address(this), totalOperatorSets++);
+
+        CreateSetParams[] memory p = CreateSetParams({operatorSetId: operatorSet.id, strategies: strategies}).toArray();
+
+        print.createOperatorSets(p);
+        allocationManager.createOperatorSets(p);
+    }
+
+    function slashOperator(User operator, uint32 operatorSetId, uint256 wadToSlash) public createSnapshot {
+        print.method("slashOperator");
+
+        SlashingParams memory p = SlashingParams({
+            operator: address(operator),
+            operatorSetId: operatorSetId,
+            wadToSlash: wadToSlash,
+            description: "bad operator"
+        });
+
+        print.slashOperator(p);
+        allocationManager.slashOperator(p);
+    }
+
+    function deregisterFromOperatorSets(User operator, uint32[] memory operatorSetIds) public createSnapshot {
+        print.method("deregisterFromOperatorSets");
+
+        DeregisterParams memory p =
+            DeregisterParams({operator: address(operator), avs: address(this), operatorSetIds: operatorSetIds});
+
+        print.deregisterFromOperatorSets(p);
+        allocationManager.deregisterFromOperatorSets(p);
+    }
+
+    function setAVSRegistrar(
+        IAVSRegistrar registrar
+    ) public createSnapshot {
+        print.method("setAVSRegistrar");
+        console.log("Setting AVS registrar to: %s", address(registrar));
+        allocationManager.setAVSRegistrar(registrar);
+    }
+
+    function addStrategiesToOperatorSet(uint32 operatorSetId, IStrategy[] memory strategies) public createSnapshot {
+        print.method("addStrategiesToOperatorSet");
+
+        console.log("Adding strategies to operator set: %d", operatorSetId);
+
+        for (uint256 i; i < strategies.length; ++i) {
+            console.log("   strategy: %s", address(strategies[i]));
+        }
+
+        allocationManager.addStrategiesToOperatorSet(operatorSetId, strategies);
+    }
+
+    function removeStrategiesFromOperatorSet(
+        uint32 operatorSetId,
+        IStrategy[] memory strategies
+    ) public createSnapshot {
+        print.method("removeStrategiesFromOperatorSet");
```

Comment:
What's the difference between using print & console.log? Can we write it in README for integration tests?

Reply:
fixed in [9413387](https://github.com/Layr-Labs/eigenlayer-contracts/pull/894/commits/9413387a75915cc60b4f91a4f9085bc8acc0a363)